nuclear/roles/CMakeLists.txt

```diff
@@ -22,14 +22,16 @@ foreach(rolefile ${rolefiles})
   # Check it is a .role file
   if(roleext STREQUAL .role)
 
-    # Get role name and add it to our list of roles
-    get_filename_component(role_name ${rolefile} NAME_WE)
+    # Roles may be within subfolders - get the subfolder and name
+    file(RELATIVE_PATH role_folder ${role_source_dir} ${rolefile})
```

Comment:
Have you considered the case where a role is not in a subfolder? What about nested subfolders?
I feel like `role_folder` is slightly misleading as a variable name, since it's the relative path, which includes the role file's name, right? In the case where it's not in a subfolder, it wouldn't contain any folder names at all, if I'm not mistaken.

Reply:
`role_path` might alleviate @KipHamiltons confusion?